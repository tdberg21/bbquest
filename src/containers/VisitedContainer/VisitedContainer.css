.visited-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  padding: 30px;
}

.visited-card {
  width: 28%;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin: 1rem;
  border: 1px solid #333;
  padding: 10px 20px;
}

.text-holder {
  display: flex;
  justify-content: space-between;
  width: 80%;
  margin: auto;
}

.visited-header {
  margin: auto;
}

.joint-image {
  width: 125px;
  margin: 10px auto;
}