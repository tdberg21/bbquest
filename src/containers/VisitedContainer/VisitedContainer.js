import React, { Component } from 'react';
import { connect } from 'react-redux';
import './VisitedContainer.css';

export class VisitedContainer extends Component {

  displayVisited = () => {
    return this.props.visited.map((joint, index) => {
      return (
        <div 
          className='visited-card' 
          key={`${index}+ ${joint.restaurant_name}`}
        >
          <h3 className='visited-header'>{joint.restaurant_name}</h3>
          <div className='text-holder'>
            <p> Rating: </p>
            <p>{joint.rating}</p>
          </div>
          <div className='text-holder'>
            <p> Date Visited:</p> 
            <p>{joint.date}</p>
          </div>
          <div className='text-holder'>
            <p> Meal: </p>
            <p>{joint.meal} </p>
          </div>
          <div className='text-holder'>
            <p> Notes: </p> 
            <p>{joint.notes} </p>
          </div>
        </div>
      );
    });
  }
  render () {
    return (
      <div className='visited-container'>
        {this.displayVisited()} 
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  visited: state.visited
});

export default connect(mapStateToProps, null)(VisitedContainer);