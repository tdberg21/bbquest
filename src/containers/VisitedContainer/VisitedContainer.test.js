import React from 'react';
import { shallow } from 'enzyme';
import { VisitedContainer } from './VisitedContainer';

describe('VISITED CONTAINER TESTS', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<VisitedContainer visited='[]'/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should return an array of divs to display', () => {
    const mockVisited = [{restaurant_name: 'PeeWees'}];
    const wrapper = shallow(<VisitedContainer visited={mockVisited} />);
    const results = wrapper.instance().displayVisited();
    expect(results.length).toEqual(1);
  });
});