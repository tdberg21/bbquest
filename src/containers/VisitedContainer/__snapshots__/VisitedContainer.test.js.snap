// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VISITED CONTAINER TESTS should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VisitedContainer
    visited={
        Array [
            Object {
              "restaurant_name": "PeeWees",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="visited-card"
>
          <h3
                    className="visited-header"
          >
                    PeeWees
          </h3>
          <img
                    className="joint-image"
                    src={undefined}
          />
          <div
                    className="text-holder"
          >
                    <p>
                               
                              <b>
                                         Rating: 
                              </b>
                               
                    </p>
                    <p />
          </div>
          <div
                    className="text-holder"
          >
                    <p>
                              <b>
                                         Date Visited:
                              </b>
                    </p>
                    <p />
          </div>
          <div
                    className="text-holder"
          >
                    <p>
                              <b>
                                         Meal: 
                              </b>
                    </p>
                    <p>
                               
                    </p>
          </div>
          <div
                    className="text-holder"
          >
                    <p>
                               
                              <b>
                                        Notes: 
                              </b>
                    </p>
                    <p>
                               
                    </p>
          </div>
</div>,
      ],
      "className": "visited-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0+ PeeWees",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="visited-header"
>
              PeeWees
</h3>,
            <img
              className="joint-image"
              src={undefined}
/>,
            <div
              className="text-holder"
>
              <p>
                             
                            <b>
                                           Rating: 
                            </b>
                             
              </p>
              <p />
</div>,
            <div
              className="text-holder"
>
              <p>
                            <b>
                                           Date Visited:
                            </b>
              </p>
              <p />
</div>,
            <div
              className="text-holder"
>
              <p>
                            <b>
                                           Meal: 
                            </b>
              </p>
              <p>
                             
              </p>
</div>,
            <div
              className="text-holder"
>
              <p>
                             
                            <b>
                                          Notes: 
                            </b>
              </p>
              <p>
                             
              </p>
</div>,
          ],
          "className": "visited-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PeeWees",
              "className": "visited-header",
            },
            "ref": null,
            "rendered": "PeeWees",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "joint-image",
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                   
                  <b>
                                     Rating: 
                  </b>
                   
</p>,
                <p />,
              ],
              "className": "text-holder",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <b>
                       Rating: 
</b>,
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " Rating: ",
                    },
                    "ref": null,
                    "rendered": " Rating: ",
                    "type": "b",
                  },
                  " ",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  <b>
                                     Date Visited:
                  </b>
</p>,
                <p />,
              ],
              "className": "text-holder",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <b>
                     Date Visited:
</b>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Date Visited:",
                  },
                  "ref": null,
                  "rendered": " Date Visited:",
                  "type": "b",
                },
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  <b>
                                     Meal: 
                  </b>
</p>,
                <p>
                   
</p>,
              ],
              "className": "text-holder",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <b>
                     Meal: 
</b>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Meal: ",
                  },
                  "ref": null,
                  "rendered": " Meal: ",
                  "type": "b",
                },
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  " ",
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                   
                  <b>
                                    Notes: 
                  </b>
</p>,
                <p>
                   
</p>,
              ],
              "className": "text-holder",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <b>
                      Notes: 
</b>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Notes: ",
                    },
                    "ref": null,
                    "rendered": "Notes: ",
                    "type": "b",
                  },
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  " ",
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="visited-card"
>
            <h3
                        className="visited-header"
            >
                        PeeWees
            </h3>
            <img
                        className="joint-image"
                        src={undefined}
            />
            <div
                        className="text-holder"
            >
                        <p>
                                     
                                    <b>
                                                 Rating: 
                                    </b>
                                     
                        </p>
                        <p />
            </div>
            <div
                        className="text-holder"
            >
                        <p>
                                    <b>
                                                 Date Visited:
                                    </b>
                        </p>
                        <p />
            </div>
            <div
                        className="text-holder"
            >
                        <p>
                                    <b>
                                                 Meal: 
                                    </b>
                        </p>
                        <p>
                                     
                        </p>
            </div>
            <div
                        className="text-holder"
            >
                        <p>
                                     
                                    <b>
                                                Notes: 
                                    </b>
                        </p>
                        <p>
                                     
                        </p>
            </div>
</div>,
        ],
        "className": "visited-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0+ PeeWees",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="visited-header"
>
                PeeWees
</h3>,
              <img
                className="joint-image"
                src={undefined}
/>,
              <div
                className="text-holder"
>
                <p>
                                 
                                <b>
                                                 Rating: 
                                </b>
                                 
                </p>
                <p />
</div>,
              <div
                className="text-holder"
>
                <p>
                                <b>
                                                 Date Visited:
                                </b>
                </p>
                <p />
</div>,
              <div
                className="text-holder"
>
                <p>
                                <b>
                                                 Meal: 
                                </b>
                </p>
                <p>
                                 
                </p>
</div>,
              <div
                className="text-holder"
>
                <p>
                                 
                                <b>
                                                Notes: 
                                </b>
                </p>
                <p>
                                 
                </p>
</div>,
            ],
            "className": "visited-card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PeeWees",
                "className": "visited-header",
              },
              "ref": null,
              "rendered": "PeeWees",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "joint-image",
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                     
                    <b>
                                         Rating: 
                    </b>
                     
</p>,
                  <p />,
                ],
                "className": "text-holder",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      <b>
                         Rating: 
</b>,
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": " Rating: ",
                      },
                      "ref": null,
                      "rendered": " Rating: ",
                      "type": "b",
                    },
                    " ",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    <b>
                                         Date Visited:
                    </b>
</p>,
                  <p />,
                ],
                "className": "text-holder",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <b>
                       Date Visited:
</b>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " Date Visited:",
                    },
                    "ref": null,
                    "rendered": " Date Visited:",
                    "type": "b",
                  },
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    <b>
                                         Meal: 
                    </b>
</p>,
                  <p>
                     
</p>,
                ],
                "className": "text-holder",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <b>
                       Meal: 
</b>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " Meal: ",
                    },
                    "ref": null,
                    "rendered": " Meal: ",
                    "type": "b",
                  },
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      undefined,
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                    " ",
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                     
                    <b>
                                        Notes: 
                    </b>
</p>,
                  <p>
                     
</p>,
                ],
                "className": "text-holder",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      <b>
                        Notes: 
</b>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Notes: ",
                      },
                      "ref": null,
                      "rendered": "Notes: ",
                      "type": "b",
                    },
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      undefined,
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                    " ",
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
